sqlmesh/core/scheduler.py

```diff
@@ -431,6 +438,37 @@ def compute_interval_params(
         ignore_cron=ignore_cron,
         end_bounded=end_bounded,
     ).items():
+
+        def intersect(left: SnapshotInterval, intervals: list[Interval]) -> SnapshotInterval | None:
+            """Returns the intersection of `left` against `intervals`. Returns None if it does not intersect. Assumes `intervals` are disjoint."""
```

Comment:
isn't a SnapshotInterval an Interval, why are the types different?

Reply:
list[Interval] is a Batch

Reply:
can we define this lambda outside of the for loop, the nestedness is a bit too much

Reply:
> isn't a SnapshotInterval an Interval, why are the types different?

From what I can tell, `missing_intervals` (defined [here](https://github.com/TobikoData/sqlmesh/blob/910261867a7cf5c8f9fbb3b2f4349f37dd348d2f/sqlmesh/core/snapshot/definition.py#L1572-L1581)) returns `sqlmesh.core.snapshot.Intervals` which is defined [here](https://github.com/TobikoData/sqlmesh/blob/910261867a7cf5c8f9fbb3b2f4349f37dd348d2f/sqlmesh/core/snapshot/definition.py#L50-L51) as `list[t.Tuple[int,int]]`. The built-in scheduler has its own `Interval` defined [here](https://github.com/TobikoData/sqlmesh/blob/910261867a7cf5c8f9fbb3b2f4349f37dd348d2f/sqlmesh/core/scheduler.py#L38-L39).

So `Interval = (datetime, datetime)` and `SnapshotInterval = (int, int)`.

Please see https://github.com/TobikoData/sqlmesh/pull/2822/commits/e9d9119249d07679bc4c04c913d4563ffdf8aef7 where we convert back and forth. Doesn't seem great, but it's unclear to me what an alternative might be.